src/app/cli/src/init/client.ml

```diff
@@ -643,6 +726,80 @@ let cancel_transaction =
              eprintf "could not deserialize user command\n" ;
              exit 16 ))
 
+let cancel_transaction_graphql =
+  let txn_id_flag =
+    Command.Param.(
+      flag "id" ~doc:"ID Transaction ID to be cancelled"
+        (required Cli_lib.Arg_type.user_command))
+  in
+  Command.async
+    ~summary:
+      "Cancel a transaction -- this submits a replacement transaction with a \
+       fee larger than the cancelled transaction."
+    (Cli_lib.Background_daemon.init ~rest:true txn_id_flag
+       ~f:(fun rest_port user_command ->
+         let receiver =
+           match
+             User_command.Payload.body (User_command.payload user_command)
+           with
+           | Payment payment ->
+               payment.receiver
+           | Stake_delegation (Set_delegate {new_delegate}) ->
+               new_delegate
+         in
+         let open Deferred.Let_syntax in
+         let%bind nonce_response =
+           let open Graphql_client.Encoders in
+           Graphql_client.query
+             (Graphql_queries.Get_inferred_nonce.make
+                ~public_key:(public_key (User_command.sender user_command))
+                ())
+             rest_port
+         in
+         let maybe_inferred_nonce =
+           let open Option.Let_syntax in
+           let%bind wallet = nonce_response#wallet in
+           let%map nonce = wallet#inferredNonce in
+           int_of_string nonce
+         in
+         let cancelled_nonce =
+           Coda_numbers.Account_nonce.to_int (User_command.nonce user_command)
+         in
+         let inferred_nonce =
+           Option.value maybe_inferred_nonce ~default:cancelled_nonce
+         in
+         let cancel_fee =
```

Comment:
I feel like we should do this logic in the graphql API layer; but we can do that as a follow up later

Reply:
yeah maybe. I could also imagine a more complicated workflow that would suggest a cancel fee to you but not enforce it